### PR TITLE
Some refine for RSOperator

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233 

Problem Summary:

### What is changed and how it works?

1. move `children` in `RSOperator` to `LogicalOp`
2. remove useless `null_direction`
3. remove useless function `switchDirection()` and `optimize()`
4. change `getAttrs()` to `getColumnIDs()` sine only column id is needed
5. use fmt to rewrite `toDebugString()`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/Filter/Unsupported.h [hunk 1]
```diff
@@ -37,14 +37,16 @@ class Unsupported : public RSOperator
...
-        return R"({"op":")" + name() + //
-            R"(","reason":")" + reason + //
-            R"(","content":")" + content + //
-            R"(","is_not":")" + DB::toString(is_not) + "\"}";
+        return fmt::format(
+            R"({{"op":"{}","reason":"{}","content":"{}","is_not":"{}"}})",
+            name(),
+            reason,
+            content,
+            DB::toString(is_not));
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Filter/Unsupported.h:49]:
```suggestion
            is_not);
```

**Review by JaySon-Huang — APPROVED** [review]
The rest LGTM

File: dbms/src/Storages/DeltaMerge/Filter/Unsupported.h (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Filter/Unsupported.h:49]: seems `is_not` is not used, can we remove directly?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Filter/Unsupported.h]: removed

File: dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp [hunk 2]
```diff
@@ -2132,7 +2114,7 @@ try
 }
 CATCH
 
-TEST_F(DMMinMaxIndexTest, ParseIn)
+TEST_F(c, ParseIn)
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp:2117]:
```suggestion
TEST_F(DMMinMaxIndexTest, ParseIn)
```

**Pull request merged**